### ISSUE TITLE
508-defect-1 ❗ Launchblocker [SCREENREADER] When adding a question, screen readers receive no feedback

### ISSUE THREAD

**Issue by AngelaFowler82** [department-of-veterans-affairs/va.gov-team#24592]
# [508-defect-1](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

Issue Title: Adding question, no indication of screen change.
Point of Contact: Angela Fowler
**VFS Point of Contact:** _First name only: Angela 

## User Story or Problem Statement:
As a screen reader user, I expect some indication as to whether clicking a link or button changed the screen.

## Details
On step two of the questionnaire, when I choose the button to add an additional question, JAWS gives no indication that the screen changed. I have to explore the screen to locate the edit box in which I can type the additional question. This could be very confusing for beginner screen reader users and those with cognitive disabilities.

## Demo of issues

https://user-images.githubusercontent.com/14154792/118188295-251afe80-b40e-11eb-9f0d-2b501bec3fb0.mov

### Relevant issues to note in the demo
- Missing error validation
- (1:13) Lack of focus management means screen reader users don't know if the "add questions" button has been activated and can result in them pressing it many times over 
- Content's `h2` is "Prepare for your appointment" which is awkwardly followed by another `h2` "What's the reason for your visit? Routine or follow-up visit" that is then followed by questions unrelated to providing reasons for a visit "Are there any additional details you’d like to share with your provider about this appointment?" (Since the user is prompted to provide **additional** details, not primary details or reasons, the language is confusing). @peggygannon 

There's another bug where the initial button is inactive as well. I'm unsure if that was intentional.
![image](https://user-images.githubusercontent.com/14154792/118189549-ed14bb00-b40f-11eb-9b1a-107b68333a71.png)

## Steps to Recreate
<ul>
<li> Access https://staging.va.gov/?next=%2Fhealth-care%2Fhealth-questionnaires%2Fquestionnaires</li>
<li>Sign in using va.api.user+idme.101@gmail.com</li>
<li> Proceed to step two of the questionnaire.</li>
<li> Using JAWS or NVDA, press "enter" on the option to add a question.</li>
</ul>

## Proposed Solution
**Josh:** The key issue identified in this ticket by Angela is caused by a lack of focus management and error states which stems from the use of the list loop pattern. An alternative, more accessible, and less magic solution to this could be to simply place "answer questions" on a separate page as an open text field.

## Acceptance Criteria (Needs review by content and design)

- [ ] List loop pattern for additional questions is replaced with an open text field with new label suggested by content team @peggygannon 
- [ ] Additional question text field is placed on a separate page with `h2` "Step 3 of 4: [descriptive title]"
- [ ] If needed, hint text is provided using `aria-describedby` to instruct the user to order questions by importance
- [ ] When the user navigates to page "Step 3 of 4:", focus should be placed on the `h2` "Step 3 of 4:"
- [ ] Step 2 of 3 is updated to Step 2 of 4
- [ ] Second `h2` "What's the reason for your visit? Routine or follow-up visit" is removed from Step 2 of 4
- [ ] Step 2 of 4's title is updated to be more descriptive @peggygannon 

## Screenshots
<img width="889" alt="Screen Shot 2021-05-13 at 4 47 09 PM" src="https://user-images.githubusercontent.com/14154792/118187070-7c1fd400-b40c-11eb-9c6d-f5b8ac74cea6.png">
<img width="1149" alt="Screen Shot 2021-05-13 at 4 55 45 PM" src="https://user-images.githubusercontent.com/14154792/118187073-7d510100-b40c-11eb-8ce9-12e3b97adb66.png">

## Demo of Step 3 of 4

https://user-images.githubusercontent.com/14154792/118187176-9d80c000-b40c-11eb-9798-69f67fb3b53b.mov

**Comment by joshkimux** [comment]
@rtwell @AngelaFowler82 @peggygannon @kristenmcconnell @leahkeeler @mdewey for awareness and review.

I've written up new acceptance criteria and a possible design/code solution to fixing this launch blocker (and a number of other defect-level 2 tickets along the way). Unless there are strong design, content, or technical reasons to use the list loop pattern in this instance, opting for a more low-tech (albeit boring) solution through an open text field on a separate page may be a more low-lift, accessible step forwards that can be completed in a weeks time (to make way for launch). Do let me know if I'm missing any important context behind these design decisions, and I can either tweak the recommended solution or focus on remediating the list loop pattern itself.

In the scenario that we do need to keep the list loop pattern, we'll need to write more complicated remediation acceptance criteria which will include unique error states, focus management, possible aria alerts, and fixing the review page (which currently seems to have a unique method of handling the list loop's submissions).

If anyone else has other ideas or alternatives, do let me know! 🙇

**Comment by mdewey** [comment]
@joshkimux  Just to be sure you are in the loop, we have made the choice to opt for the textarea. I will tag you once its in staging for you to review :-)

**Comment by joshkimux** [comment]
@mdewey Thank you for the update and thank you to the team for the fast pivot!

**Comment by mdewey** [comment]
![screencapture-localhost-3001-health-care-health-questionnaires-questionnaires-answer-questions-review-and-submit-2021-05-18-09_56_58](https://user-images.githubusercontent.com/1793923/118664412-7258f080-b7bf-11eb-9cd3-c6239f5be8ec.png)
![screencapture-localhost-3001-health-care-health-questionnaires-questionnaires-answer-questions-prepare-for-your-appointment-2021-05-18-09_56_36](https://user-images.githubusercontent.com/1793923/118664418-738a1d80-b7bf-11eb-9074-fd7addfed181.png)


@kristenmcconnell  Here is what the UI looks like with the textarea. I am still working through some of the technical updates, but I wanted your feedback sooner rather than later. Let me know what you think.

**Comment by kristenmcconnell** [comment]
We need to adjust the wording slightly. Right now, it looks like we only want them to enter one question. Lets change it to...
"Do you have any questions you want to ask your provider? Please enter your most important question first."

**Comment by joshkimux** [comment]
@mdewey @kristenmcconnell will we be able to break apart these questions onto separate pages, or are we saving that for a future state?

**Comment by mdewey** [comment]
Since the one page tested well (@kristenmcconnell can speak more to that), we planned on saving that for a future state.

**Comment by kristenmcconnell** [comment]
@mdewey is correct. Since it did test well, then we are going to save that part for a future state.

**Comment by joshkimux** [comment]
Got it! Just for clarification, the why behind this ask is specifically for screen reader users who navigate by headings. The current state's heading structure is a bit confusing if viewed as a table of contents of the page.

1. Answer primary care questionnaire
   2. Step 2 of 3: Prepare for your appointment
   2. What's the reason for your visit? Routine or follow up visit.

Folks who are checking the headings list first with a screenreader may think that there are questions beneath "Step 2 of 3..." and that "What's the reason for your visit. Routine or follow up visit." will include a question that allows them to pick between a routine or follow up visit (although I'm guessing they are the same thing).

Confusion may compound when:
- they realize that Step 2 of 3 has no direct children or questions nested under it
- the questions under "What's the reason for your visit. Routine or follow up visit." don't include a question on picking between a routine or follow up visit, and contains questions unrelated to the heading "additional details" being different than "reason"
- additional questions is nested under "What's the reason for your visit" 

Those could be addressed by changing the heading structure to:
1. Answer primary care questionnaire
   2. Step 2 of 4: Provide details for your visit
[Complete step] 

1. Answer primary care questionnaire
   2. Step 3 of 4: Add questions for your provider

**Comment by mdewey** [comment]
@joshkimux  Just a heads up,  we are removing `What's the reason for your visit? Routine or follow up visit.` as a header to avoid the screen reader confusion.

**Comment by joshkimux** [comment]
Awesome! Thank you for the heads up

**Comment by mdewey** [comment]
@joshkimux The removal of the list loop is in staging. This addresses the feedback in the ticket.

**Comment by AngelaFowler82** [comment]
Looks good. The perfect way to end a Friday! Unless Josh has anything, I think we can close this.

**Comment by kristenmcconnell** [comment]
As Angela said, it looks good.

**Comment by joshkimux** [comment]
Closing out! 🥳 
Big kudos to the team for getting this fixed so quickly
